mesonbuild/backend/backends.py

```diff
@@ -316,6 +316,20 @@ def get_target_filename(self, t: T.Union[build.Target, build.CustomTargetIndex],
     def get_target_filename_abs(self, target: T.Union[build.Target, build.CustomTargetIndex]) -> str:
         return os.path.join(self.environment.get_build_dir(), self.get_target_filename(target))
 
+    def get_target_debug_filename(self, target: build.BuildTarget) -> str:
+        assert isinstance(target, build.BuildTarget), target
+        if target.get_debug_filename():
+            debug_filename = target.get_debug_filename()
+            return os.path.join(self.get_target_dir(target), debug_filename)
+        else:
+            return None
+
+    def get_target_debug_filename_abs(self, target: build.BuildTarget) -> str:
+        assert isinstance(target, build.BuildTarget), target
+        if not target.get_debug_filename():
+            return None
+        return os.path.join(self.environment.get_build_dir(), self.get_target_debug_filename(target))
```

Comment:
The return types on these are both wrong, this should be `-> T.Optional[str]`, since it can return `None`

Reply:
Good catch.